distribution/lib/Standard/Base/0.0.0-dev/src/Data/Number/Internal.enso

```diff
@@ -0,0 +1,1041 @@
+       syntax.
+
+       Arguments:
+       - fmt: The java-style formatting specifier.
+
+       > Example
+         Convert the value 5 to a string.
+
+             5.format "%x"
+    format : Text -> Text
+    format fmt = String.format fmt this
+
+    ## Checks equality of numbers, using an `epsilon` value.
+
+       Arguments:
+       - that: The number to check equality against.
+       - epsilon: The value by which `this` and `that` can be separated by before
+         counting as not equal.
+
+       > Example
+         Check if 1 is equal to 1.0000001 within 0.001.
+
+             1.equals 1.0000001 epsilon=0.001
+    equals : Number -> Number -> Boolean
+    equals that epsilon=0.0 =
+        (this == that) || ((this - that).abs <= epsilon)
+
+    ## Returns the smaller value of `this` and `that`.
+
+       Arguments:
+       - that: The number to compare `this` against.
+
+       ? Math.min or Number.min
+         While we provide the min method on `Number`, we find it more intuitive to
+         write `Math.min a b` rather than `a.min b`. To that end, we recommend using
+         the first style.
+
+       > Example
+         Find the minimum of 2 and 5.
+
+             2.min 5
+    min : Number -> Number
+    min that = if this < that then this else that
+
+    ## Returns the larger value of `this` and `that`.
+
+       Arguments:
+       - that: The number to compare `this` against.
+
+       ? Math.max or Number.max
+         While we provide the max method on `Number`, we find it more intuitive to
+         write `Math.max a b` rather than `a.max b`. To that end, we recommend using
+         the first style.
+
+       > Example
+         Find the maximum of 2 and 5.
+
+             2.max 5
+    max : Number -> Number
+    max that = if this > that then this else that
+
+    ## Number to JSON conversion.
+
+       > Example
+         Convert the number 8 to JSON.
+
+             8.to_json
+    to_json : Json.Number
+    to_json = Json.Number this
+
+    ## A constant holding the floating-point positive infinity.
+    positive_infinity : Decimal
+    positive_infinity = Double.POSITIVE_INFINITY
+
+    ## A constant holding the floating-point negative infinity.
+    negative_infinity : Decimal
+    negative_infinity = Double.NEGATIVE_INFINITY
+
+    ## A constant holding the floating-point Not-a-Number value.
+    nan : Decimal
+    nan = Double.NaN
+
+    ## Checks if the given number is the floating-point Not-a-Number value.
+
+       This is needed, because the NaN value will return `False` even when being
+       compared with itself, so `x == Number.nan` would not work.
+    is_nan : Boolean
+    is_nan = case this of
+        Decimal -> Double.isNaN this
+        _ -> False
+
+    ## Returns the sign of the number.
+    signum : Integer
+    signum =
+        if this > 0 then 1 else
+            if this < 0 then -1 else 0
+
+
+###################
+## Decimal
+###################
```

Comment:
This format of comments does not seem to be used anywhere else in the `.enso` codebase. Nothing against it, just asking to ensure consistency.

Reply:
:+1: